src/test/java/software/amazon/cloudwatchlogs/emf/model/MetricDirectiveThreadSafetyTest.java

```diff
@@ -0,0 +1,75 @@
+package software.amazon.cloudwatchlogs.emf.model;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import org.junit.After;
+import org.junit.Test;
+
+public class MetricDirectiveThreadSafetyTest {
```

Comment:
does it cover all the shard fields? how about dimensions?

Reply:
The dimension-related cocurrency tests are included in the MetricsLoggerThreadSafetyTest.java. Ideally we should have all concurrency tests located in one file, but since the `MetricDirective` class is not public and I can not access its method and fields outside of its package in order to validate the test results, I have to put the putMetric()-related tests into this MetricDirectiveThreadSafetyTest.java file separately...